### PR TITLE
virttest.qemu_vm:Add a common function to control options for virtio pci device

### PR DESCRIPTION
**Pull request by yanan-fu**
There are multiple options for virtio pci device, need use different values for testing. Add a common function to control them.

id: 1542301
Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by yanan-fu** [comment]
@huangyum could you help review this PR, thanks a lot!

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1321,6 +1321,24 @@ def add_disable_modern(devices, dev, dev_type):
                 value = params.get("disable_modern", "on")
                 dev.set_param("disable-modern", value)
 
+        def add_iommu_platform(devices, dev, dev_type):
```
Comment (hereischen) [inline virttest/qemu_vm.py]: We may need to refactor these four functions as they are almost same.

Comment (yanan-fu) [reply virttest/qemu_vm.py]: updated,  use a common function instead. Could you help review again? thanks a lot!

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by jingzhao84** [comment]
Acked-by jinzhao jinzhao@redhat.com

 (1/2) smp_4.4096m.repeat1.Host_RHEL.m7.u5.spice.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1040.31 s)
 (2/2) smp_4.4096m.repeat1.Host_RHEL.m7.u5.spice.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.shutdown.q35: PASS (53.39 s)

**Comment by yanan-fu** [comment]
@luckyh  could you help review this PR, thanks a lot !

File: virttest/qemu_vm.py [hunk 2]
```diff
@@ -2302,11 +2293,17 @@ def sort_key(dev):
             # disable-legacy and disable-modern options explicitly.
             set_disable_legacy = params.get("set_disable_legacy", "no")
             set_disable_modern = params.get("set_disable_modern", "no")
+            set_iommu_platform = params.get("set_iommu_platform", "no")
+            set_ats = params.get("set_ats", "no")
             if dev_type in virtio_pci_devices:
                 if set_disable_legacy == "yes":
-                    add_disable_legacy(devices, device, dev_type)
+                    add_virtio_option("disable-legacy", "on",  devices, device, dev_type)
```
Comment (CongLi) [inline virttest/qemu_vm.py]: default value for disable-legacy should be off, right? refer line 1312

File: virttest/qemu_vm.py [hunk 3]
```diff
@@ -2302,11 +2293,17 @@ def sort_key(dev):
...
                 if set_disable_modern == "yes":
-                    add_disable_modern(devices, device, dev_type)
+                    add_virtio_option("disable-modern", "off", devices, device, dev_type)
```
Comment (CongLi) [inline virttest/qemu_vm.py]: default value for disable-modern should be off, right? refer line 1321

Comment (CongLi) [inline virttest/qemu_vm.py]: default value for disable-legacy should be on, right? refer line 1321

File: virttest/qemu_vm.py (hunk 2 above)
Comment (jingzhao84) [reply virttest/qemu_vm.py]: @CongLi  based on machine type, if you run 440fx, the default value is off

Comment (yanan-fu) [reply virttest/qemu_vm.py]: @CongLi   L1312 is the old value.   We use virtio 1.0 by default now,  so disable-legacy is on by default.

And this change will not effect other cases unless  "set_disable_legacy" be set to "yes".

File: virttest/qemu_vm.py (hunk 3 above)
Comment (yanan-fu) [reply virttest/qemu_vm.py]: @CongLi  same with disable_legacy, thanks !

File: virttest/qemu_vm.py (hunk 2 above)
Comment (xutian) [reply virttest/qemu_vm.py]: Hi guys,

here just need to care code level default values,  default value for  different machine type is should been control by machine.cfg

Thanks,
Xu

File: virttest/qemu_vm.py (hunk 3 above)
Comment (xutian) [reply virttest/qemu_vm.py]: same as above.

File: virttest/qemu_vm.py (hunk 2 above)
Comment (yanan-fu) [reply virttest/qemu_vm.py]: Thanks @xutian . yes, and  i does not distinguish the default value for different machine type here.
Only one default value here.

File: virttest/qemu_vm.py (hunk 3 above)
Comment (yanan-fu) [reply virttest/qemu_vm.py]: same as above :)

**Comment by CongLi** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Comment by yanan-fu** [comment]
@luckyh  could you help review this PR ? thanks a lot!

File: virttest/qemu_vm.py [hunk 4]
```diff
@@ -2302,11 +2293,17 @@ def sort_key(dev):
             # disable-legacy and disable-modern options explicitly.
             set_disable_legacy = params.get("set_disable_legacy", "no")
             set_disable_modern = params.get("set_disable_modern", "no")
+            set_iommu_platform = params.get("set_iommu_platform", "no")
```
Comment (luckyh) [inline virttest/qemu_vm.py]: I'd suggest using only one param to represent these features.

Given option `foo` and its value `bar`:
a. set `foo=on` if the value is `on`
b. set `foo=off` if the value is `off`
c. do not set the option if the value is undefined (`None`)

Comment (yanan-fu) [reply virttest/qemu_vm.py]: @luckyh  thanks, updated, could you help review it again ?
And the reference will be updated in another patch later.

**Comment by yanan-fu** [comment]
Test with case "boot", with following code be added in case cfg,
```
    virtio_dev_disable_legacy = on
    virtio_dev_disable_modern = off
    virtio_dev_iommu_platform = on
    virtio_dev_ats = on
```
qemu command line:
```
    -device virtio-scsi-pci,id=virtio_scsi_pci0,bus=pci.0,addr=0x4,disable-legacy=on,disable-modern=off,iommu_platform=on,ats=on \
    -drive id=drive_image1,if=none,snapshot=off,aio=threads,cache=none,format=qcow2,file=/home/kvm_autotest_root/images/rhel75-64-virtio-scsi.qcow2 \
    -device scsi-hd,id=image1,drive=drive_image1 \
    -device virtio-net-pci,mac=9a:af:b0:b1:b2:b3,id=id2Gv778,vectors=4,netdev=idJCn5CK,bus=pci.0,addr=0x5,disable-legacy=on,disable-modern=off,iommu_platform=on,ats=on  \
    -netdev tap,id=idJCn5CK,vhost=on,vhostfd=15,fd=20 \
```

**Pull request merged**